src/main/java/com/arpnetworking/tsdcore/sinks/MetricsDataSerializer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Dropbox Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.tsdcore.sinks;
+
+import com.arpnetworking.metrics.aggregation.protocol.Messages;
+import com.arpnetworking.metrics.mad.model.AggregatedData;
+import com.arpnetworking.metrics.mad.model.statistics.HistogramStatistic;
+import com.arpnetworking.metrics.mad.model.statistics.Statistic;
+import com.arpnetworking.metrics.mad.model.statistics.StatisticFactory;
+import com.arpnetworking.tsdcore.model.AggregationMessage;
+import com.arpnetworking.tsdcore.model.PeriodicData;
+import com.google.protobuf.ByteString;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Utility class to serialize to metrics-aggregator-protocol protobuf messages.
```

Comment:
nit: It doesn't actually serialize the message, it converts our internal model to the protobuf model.